File: src/main/java/upbrella/be/store/dto/response/SingleStoreIntroductionResponse.java
package upbrella.be.store.dto.response;

import lombok.Builder;
import lombok.Getter;
import upbrella.be.store.entity.StoreDetail;
import upbrella.be.store.entity.StoreMeta;

import java.util.List;
import java.util.stream.Collectors;

@Getter
@Builder
public class SingleStoreIntroductionResponse {

    private long id;
    private String thumbnail;
    private String name;
    private String category;

    public static SingleStoreIntroductionResponse of(long id, String thumbnail, String name, String category) {

        return SingleStoreIntroductionResponse.builder()
                .id(id)
                .thumbnail(thumbnail)
                .name(name)
                .category(category)
                .build();
    }
    public static SingleStoreIntroductionResponse createSingleIntroduction(StoreDetail storeDetail) {

        List<SingleImageUrlResponse> sortedImageUrls = storeDetail.getSortedStoreImages().stream()
                .map(SingleImageUrlResponse::createImageUrlResponse)
                .collect(Collectors.toList());

        String thumbnail = createThumbnail(sortedImageUrls);
        StoreMeta storeMeta = storeDetail.getStoreMeta();

        return of(storeMeta.getId(), thumbnail, storeMeta.getName(), storeMeta.getCategory());
    }

    private static String createThumbnail(List<SingleImageUrlResponse> imageUrls) {

        return imageUrls.stream()
                .findFirst()
                .map(SingleImageUrlResponse::getImageUrl)
                .orElse(null);
    }
}

File: src/main/kotlin/upbrella/be/store/dto/response/SingleStoreIntroductionResponse.kt
package upbrella.be.store.dto.response

import upbrella.be.store.entity.StoreDetail

data class SingleStoreIntroductionResponse(
    val id: Long,
    val thumbnail: String?,
    val name: String?,
    val category: String?
) {
    companion object {
        fun of(id: Long, thumbnail: String?, name: String, category: String): SingleStoreIntroductionResponse {
            return SingleStoreIntroductionResponse(id, thumbnail, name, category)
        }

        fun createSingleIntroduction(storeDetail: StoreDetail): SingleStoreIntroductionResponse {
            val sortedImageUrls = storeDetail.getSortedStoreImages()
                .map { SingleImageUrlResponse.createImageUrlResponse(it) }

            val thumbnail = createThumbnail(sortedImageUrls)
            val storeMeta = storeDetail.storeMeta

            return of(
                id = storeMeta?.id ?: throw IllegalArgumentException("ID must not be null"),
                thumbnail = thumbnail,
                name = storeMeta.name,
                category = storeMeta.category
            )
        }

        private fun createThumbnail(imageUrls: List<SingleImageUrlResponse>): String? {
            return imageUrls.firstOrNull()?.imageUrl
        }
    }
}
